src/IPAddr.cc

```diff
@@ -60,26 +60,29 @@ ConnKey::ConnKey(Val* v) {
 
     int orig_h, orig_p; // indices into record's value list
     int resp_h, resp_p;
+    uint8_t proto;
 
     if ( vr == id::conn_id ) {
         orig_h = 0;
         orig_p = 1;
         resp_h = 2;
         resp_p = 3;
+        proto = 4;
     }
     else {
         // While it's not a conn_id, it may have equivalent fields.
         orig_h = vr->FieldOffset("orig_h");
         resp_h = vr->FieldOffset("resp_h");
         orig_p = vr->FieldOffset("orig_p");
         resp_p = vr->FieldOffset("resp_p");
+        proto = vr->FieldOffset("proto");
 
         if ( orig_h < 0 || resp_h < 0 || orig_p < 0 || resp_p < 0 ) {
```

Comment:
Should this cover `proto < 0` too?

Reply:
`proto` is unsigned, so it can't ever be < 0.

Reply:
Actually Tim, I misread the code higher up. None of this is about the value of the proto, it's about checking the RecordVal's fields. In line 63, proto should also be an int, and the `FieldOffset()` check can then theoretically be negative for it. And we're assigning 4 above, not a proto value.

Reply:
Fixed